Lib/test/test_capi/test_sys.py

```diff
@@ -211,6 +206,74 @@ def test_sys_writestderr(self):
         # Test PySys_WriteStderr()
         self._test_sys_writestream('PySys_WriteStderr', 'stderr')
 
+    @unittest.skipIf(_testlimitedcapi is None, 'need _testlimitedcapi module')
+    def test_sys_audit(self):
+        # Test PySys_Audit()
+        sys_audit = _testlimitedcapi.sys_audit
+        audit_events = []
+        def audit_hook(event, args):
+            audit_events.append((event, args))
+            return None
+
+        sys.addaudithook(audit_hook)
+        try:
+            result = sys_audit("test.event", "OO", 1, "a")
+            self.assertEqual(result, 0)
+            self.assertEqual(audit_events[-1][0], "test.event")
+            self.assertEqual(audit_events[-1][1], (1, "a"))
+
+            result = sys_audit("test.no_args", "")
+            self.assertEqual(result, 0)
+            self.assertEqual(audit_events[-1][0], "test.no_args")
+            self.assertEqual(audit_events[-1][1], ())
+
+            with self.assertRaises(TypeError):
+                sys_audit(123, "O", 1)
+
+            result = sys_audit("テスト.イベント", "O", 42)
+            self.assertEqual(result, 0)
+            self.assertEqual(audit_events[-1][0], "テスト.イベント")
+            self.assertEqual(audit_events[-1][1], (42,))
+
+            with self.assertRaises(UnicodeDecodeError):
+                sys_audit(b"test.non_utf8\xff", "O", 1)
+        finally:
```

Comment:
Add also tests for `sys_audit("test.event", "(")`, `sys_audit("test.event", "&")`, `sys_audit("test.event", b"\xff")` and `sys_audit("test.event", "{OO}", [], [])`.

Add also tests for `sys_audit(NULL, "O", 1)` and `sys_audit("test.event", NULL)`. If they crash, just add comments.